jsx/Form.js

```diff
@@ -1009,61 +1020,72 @@ class DateElement extends Component {
+  }
+
   handleChange(e) {
-    this.props.onUserInput(this.props.name, e.target.value, e.target.id, 'date');
+    this.props.onUserInput(this.props.name, e.target.value);
   }
 
   render() {
     let disabled = this.props.disabled ? 'disabled' : null;
     let required = this.props.required ? 'required' : null;
     let requiredHTML = null;
+    let errorMessage = null;
+    let elementClass = null;
 
     // Add required asterix
     if (required) {
       requiredHTML = <span className="text-danger">*</span>;
     }
 
-    // Check if props minYear and maxYear are valid values if supplied
-    let minYear = this.props.minYear;
-    let maxYear = this.props.maxYear;
-    if (this.props.minYear === '' || this.props.minYear === null) {
-      minYear = '1000';
-    }
-    if (this.props.maxYear === '' || this.props.maxYear === null) {
-      maxYear = '9999';
+    // Add error message
+    if (this.props.errorMessage) {
+      errorMessage = <span>{this.props.errorMessage}</span>;
+      elementClass = 'has-error';
     }
 
-    // Handle date format
-    let format = this.props.dateFormat;
-    let inputType = 'date';
-    let minFullDate = minYear + '-01-01';
-    let maxFullDate = maxYear + '-12-31';
-    if (!format.match(/d/i)) {
-      inputType = 'month';
-      minFullDate = minYear + '-01';
-      maxFullDate = maxYear + '-12';
-    }
+    const todayButton = this.props.today && (
+      <Button
+        label='Today'
+        type="button"
+        onClick={this.handleButton}
+      />
+    );
+
+    const label = this.props.label && (
+      <label className="control-label" htmlFor={this.props.id}>
+        {this.props.label}
+        {requiredHTML}
+      </label>
+    );
 
     return (
-      <div className="row form-group">
-        <label className="col-sm-3 control-label" htmlFor={this.props.label}>
-          {this.props.label}
-          {requiredHTML}
-        </label>
-        <div className="col-sm-9">
-          <input
-            type={inputType}
-            className="form-control"
-            name={this.props.name}
-            id={this.props.id}
-            min={minFullDate}
-            max={maxFullDate}
-            onChange={this.handleChange}
-            value={this.props.value || ''}
-            required={required}
-            disabled={disabled}
-          />
-        </div>
+      <div className={elementClass}>
+        {label}
+        <InlineField weights={[1, 0]}>
+          <>
+            <input
+              type="date"
+              className="form-control"
+              name={this.props.name}
+              id={this.props.id}
+              min={this.props.minYear}
+              max={this.props.maxYear}
+              onChange={this.handleChange}
+              value={this.props.value || ''}
+              required={required}
+              disabled={disabled}
+            />
+            {errorMessage}
+          </>
+          {todayButton}
+        </InlineField>
```

Comment:
not in **24.1**

_added to improve form functionality and UI + adding today button_

to include as **CBIG override** with plans to push to **main** in some capacity